staging/src/k8s.io/kubectl/pkg/cmd/set/set_env_test.go

```diff
@@ -79,7 +79,7 @@ func TestSetEnvLocal(t *testing.T) {
 }
 
 func TestSetEnvLocalNamespace(t *testing.T) {
-	tf := cmdtesting.NewTestFactory().WithNamespace("test")
+	tf := cmdtesting.NewTestFactory()
```

Comment:
Did you mean to change this? I don't think you made any changes to set_env.go, so I wonder if this can be left unchanged.

Reply:
Yes, actually this is used to pass the test.

Reply:
I don't understand. You didn't change anything in set_env.go, so how can this test need to be changed? 

Isn't `tf := cmdtesting.NewTestFactory().WithNamespace("test")` still a valid test case that should pass?

Reply:
OK, maybe I see... is it because you changed config_flags_fake?

So now in order to get it to return false as before, you need to set namespace to empty string.

Reply:
Because I changed :
```
func (c *namespacedClientConfig) Namespace() (string, bool, error) {
	return c.namespace, len(c.namespace) > 0, nil
}
```
in `staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags_fake.go`. And the test need the namespace to be `existing-ns`. But with my change, it will have `test` as its namespace because now `enfocenamespace == true`.

Reply:
Yes, exactly :)